pkgs/build-support/rust/hooks/cargo-check-hook.sh

```diff
@@ -0,0 +1,41 @@
+declare -a checkFlags
+
+cargoCheckHook() {
+    echo "Executing cargoCheckHook"
+
+    runHook preCheck
+
+    if [[ -n "${buildAndTestSubdir-}" ]]; then
+        pushd "${buildAndTestSubdir}"
+    fi
+
+    if [[ -z ${dontUseCargoParallelTests-} ]]; then
+        threads=$NIX_BUILD_CORES
+    else
+        threads=1
+    fi
+
+    argstr="--${cargoBuildType} --target @rustTargetPlatformSpec@ --frozen";
```

Comment:
Seems like you changed:

* `${lib.optionalString (checkType == "release") "--release"}` ->  `--${cargoBuildType}`

But there are crates out there that on `checkType`, like `cargo-deb`. There may be others too.

Reply:
This is an oversight. We should pass `checkType` as `cargoCheckType` and then use that in the hook.

Reply:
I don't know how many packages are out there using that, but I added this to work around issues with packages where `cargo test` only works with `debug`, so this shouldn't be dropped entirely IMHO.

Reply:
I agree, it's an oversight. I'll do a PR that fixes this.

Reply:
@bobrik @Ma27 fix in #114455